src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedNewTests.cs

```diff
@@ -0,0 +1,928 @@
+                // (23,19): error CS9366: The type 'int[]' may not be used as the target-type of 'new'.
+                //         var v12 = (int[])new();
+                Diagnostic(ErrorCode.ERR_BadTargetTypeForNew, "(int[])new()").WithArguments("int[]").WithLocation(23, 19),
+                // (24,20): error CS0246: The type or namespace name 'Error' could not be found (are you missing a using directive or an assembly reference?)
+                //         var v13 = (Error)new();
+                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "Error").WithArguments("Error").WithLocation(24, 20),
+                // (25,19): error CS9366: The type 'T' may not be used as the target-type of 'new'.
+                //         var v14 = (T)new();
+                Diagnostic(ErrorCode.ERR_BadTargetTypeForNew, "(T)new()").WithArguments("T").WithLocation(25, 19),
+                // (12,13): warning CS0219: The variable 'v1' is assigned but its value is never used
+                //         var v1 = (Struct)new();
+                Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "v1").WithArguments("v1").WithLocation(12, 13),
+                // (18,13): warning CS0219: The variable 'v7' is assigned but its value is never used
+                //         var v7 = (int)new();
+                Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "v7").WithArguments("v7").WithLocation(18, 13),
+                // (20,13): warning CS0219: The variable 'v9' is assigned but its value is never used
+                //         var v9 = (int?)new();
+                Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "v9").WithArguments("v9").WithLocation(20, 13)
+                );
+        }
+
+        [Fact]
+        public void TestMethodInvocation()
+        {
+            var comp = CreateCompilation(@"
+using System;
+class X {
+    
+    static void M(object a, X b) => Console.Write($""{a} {b}"");
+
+    public static void Main()
+    {
+        M(new(), new());
+    }
+}
+", options: TestOptions.ReleaseExe);
+
+            CompileAndVerify(comp, expectedOutput: "System.Object X");
+        }
+
+        [Fact]
+        public void TestAmbigCall()
+        {
+            var comp = CreateCompilation(@"
+using System;
+class X {
+    
+    static void M(object a, X b) => Console.Write($""{a} {b}"");
+    static void M(X a, object b) => Console.Write($""{a} {b}"");
+
+    public static void Main()
+    {
+        M(new(), new());
+    }
+}
+").VerifyDiagnostics(
+                // (10,9): error CS0121: The call is ambiguous between the following methods or properties: 'X.M(object, X)' and 'X.M(X, object)'
+                //         M(new(), new());
+                Diagnostic(ErrorCode.ERR_AmbigCall, "M").WithArguments("X.M(object, X)", "X.M(X, object)").WithLocation(10, 9)
+                );
+        }
+
+        [Fact]
+        public void TestObjectInitializer()
+        {
+            var comp = CreateCompilation(@"
+using System;
+class X {
+    
```

Comment:
Tiny nit: the formatting in the code strings differs from the conventions we use in code.

Reply:
I will clean this up once we're addressing the test plan preferably in a followup PR - after merge we have the luxury of trying things out in sharplab.io which will help with test cases.